From oliver@neukum.org  Fri Oct  9 11:42:55 2009
From: Oliver Neukum <oliver@neukum.org>
Date: Thu, 1 Oct 2009 14:54:46 +0200
Subject: USB: visor: fix trivial accounting bug in visor driver
To: "Greg Kroah-Hartman" <gregkh@suse.de>
Message-ID: <200910011454.47038.oliver@neukum.org>
Content-Disposition: inline


usb:usbserial:visor: fix accounting in error case

data not pushed to the tty layer due to an error mustn't be counted

Signed-off-by: Oliver Neukum <oliver@neukum.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/visor.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/usb/serial/visor.c
+++ b/drivers/usb/serial/visor.c
@@ -513,7 +513,8 @@ static void visor_read_bulk_callback(str
 			tty_kref_put(tty);
 		}
 		spin_lock(&priv->lock);
-		priv->bytes_in += available_room;
+		if (tty)
+			priv->bytes_in += available_room;
 
 	} else {
 		spin_lock(&priv->lock);
